nucleus/core/logging/src/main/java/com/sun/enterprise/server/logging/GFFileHandler.java

```diff
@@ -551,68 +567,65 @@ private void configureUniformLogFormatter(String excludeFields, boolean multiLin
+            if (recordEndMarker == null || ("").equals(recordEndMarker)) {
+                recordEndMarker = RECORD_END_MARKER;
+            }
 
-        String recordFieldSeparator = manager.getProperty(cname + ".logFormatFieldSeparator");
-        if (recordFieldSeparator == null || ("").equals(recordFieldSeparator) || recordFieldSeparator.length() > 1) {
-            recordFieldSeparator = RECORD_FIELD_SEPARATOR;
-        }
+            recordFieldSeparator = manager.getProperty(cname + ".logFormatFieldSeparator");
+            if (recordFieldSeparator == null || ("").equals(recordFieldSeparator) || recordFieldSeparator.length() > 1) {
+                recordFieldSeparator = RECORD_FIELD_SEPARATOR;
+            }
 
-        String recordDateFormat = manager.getProperty(cname + ".logFormatDateFormat");
-        if (recordDateFormat != null && !("").equals(recordDateFormat)) {
-            SimpleDateFormat sdf = new SimpleDateFormat(recordDateFormat);
-            try {
-                sdf.format(new Date());
-            } catch (Exception e) {
+            recordDateFormat = manager.getProperty(cname + ".logFormatDateFormat");
+            if (recordDateFormat != null && !("").equals(recordDateFormat)) {
+                SimpleDateFormat sdf = new SimpleDateFormat(recordDateFormat);
+                try {
+                    sdf.format(new Date());
+                } catch (Exception e) {
+                    recordDateFormat = RECORD_DATE_FORMAT;
+                }
+            } else {
                 recordDateFormat = RECORD_DATE_FORMAT;
             }
-        } else {
-            recordDateFormat = RECORD_DATE_FORMAT;
-        }
 
-        formatterClass.setRecordBeginMarker(recordBeginMarker);
-        formatterClass.setRecordEndMarker(recordEndMarker);
-        formatterClass.setRecordDateFormat(recordDateFormat);
-        formatterClass.setRecordFieldSeparator(recordFieldSeparator);
+            formatterClass.setRecordBeginMarker(recordBeginMarker);
+            formatterClass.setRecordEndMarker(recordEndMarker);
+            formatterClass.setRecordDateFormat(recordDateFormat);
+            formatterClass.setRecordFieldSeparator(recordFieldSeparator);
+        }        
     }
-
+    
     void initializePump() {
-        if (pumpFuture != null) {
-            pumpFuture.cancel(true);
-            pumpFuture = null;
-        }
-        if (logToFile) {
-            pumpFuture = payaraExecutorService.submit(
-                () -> {
-                    while (!done.isSignalled() && logToFile) {
-                        try {
-                            log();
-                        } catch (Exception e) {
-                            // GLASSFISH-19125
-                            // Continue the loop without exiting
-                        }
+        pump = new Thread() {
+            @Override
+            public void run() {
+                while (!done.isSignalled()) {
```

Comment:
There also was another branch in the newer version, flushing records if `logToFile` is set to false in course of run. I believe that one should be retained